+utils/compare_symbols.m

```diff
@@ -0,0 +1,149 @@
+    end
+else
+    compiled_symbols_file = optionals.compiled_symbols_file;
+end
+if optionals.nrniv_symbols_file == ""
+    nrniv_symbols_file = fullfile(this_dir, "nrniv.txt");
+    cmd = "nm '" + utils.Paths.libnrniv_file() + "' > '" + nrniv_symbols_file + "'";
+    status = system(cmd);
+    if status
+        error("system call failed")
+    end
+else
+    nrniv_symbols_file = optionals.nrniv_symbols_file;
+end
+
+% Get the symbols lists as one big char array
+% For easier later regex search, make sure there is a newline at the start and end
+compiled_content = [10 fileread(compiled_symbols_file) 10];
+nrniv_content = [10 fileread(nrniv_symbols_file) 10];
+
+%% Parse the needed_symbols_file
+% Assume that lines starting with MANGLED
+% and NON_MANGLED have a symbol that should be checked
+possible_symbols = readlines(needed_symbols_file);
+max_nr_symbols = length(possible_symbols);
+varTypes = ["string", "logical", "string", "logical", "string"];
+varNames = ["Symbol", "is_mangled", "Interface", "is_matching", "Nrniv"];
+list_symbols = table('Size', [max_nr_symbols, length(varNames)], 'VariableTypes', varTypes, 'VariableNames', varNames);
+for i = 1:max_nr_symbols
+    line = possible_symbols(i);
+    if contains(line, "Import functions for which name mangling goes awry")
+        % TODO: something for the functions that we know are problematic,
+        % for now they need to be manually checked
+        break
+    end
+    if line.startsWith({'MANGLED', 'NON_MANGLED'})
+        parts = line.split;
+        if length(parts) ~= 3
+            error("line in file starting with MANGLED cannot be split in exactly three parts, script needs to be adapted")
+        elseif ~parts.endsWith(';')
+            error("The third part of a line in file starting with MANGLED or NON_MANGLED does not end with ;, script needs to be adapted")
+        end
+        % symbols that are still in neuron_dllimports, but are no longer
+        % used anywhere. Can possibly make this an output of this script,
+        % instead of hardcoding here
+        % if any(contains(["nrn_popsec", "hoc_call_objfunc", "hoc_obj_ref", "hoc_ret", "hoc_objectdata"], parts{3}(1:end-1)))
+        %     continue
+        % end
+        list_symbols.Symbol(i) = parts{3}(1:end-1);
+        list_symbols.is_mangled(i) = line.startsWith('MANGLED');
+    end
+end
+emptyrows = ismissing(list_symbols.Symbol);
+list_symbols = list_symbols(~emptyrows,:);
+nr_symbols = height(list_symbols);
+
+%% Try to find the list of actual symbols in both shared libraries
+% Use regex to search through full files at once
+for i = 1:nr_symbols
+    %% The symbol in our interface
+    searchstr = list_symbols.Symbol(i);
+    if isequal(searchstr, "hoc_pushs")
+        % terrible workaround
+        % because hoc_pushstr is also in the search list, will find two
+        % symbols matching hoc_pushs, use mangled name as we currently know
+        % it
+        searchstr = "_Z9hoc_pushsP6Symbol";

```

Comment:
Could possibly match on string followed by capital character or end of string instead

Reply:
Even better solution: use negative-lookahead to exclude hoc_pushstr from being found.